### PR TITLE
Prepare files under testdata/ from Makefile

### PR DESCRIPTION
**Pull request by kzys**
*Issue #, if available:*

NA

*Description of changes:*

The testdata/ directory must have vmlinux, root-drive.img and firecracker
binary, in order to run Go tests.

This change prepares the files by just running "make".

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prepare-testdata.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    local from to
+    from="$1"
+    to="$2"
+
+    if [ ! -e "$to" ]; then
+        curl -fsSL -o "$to" "$from"
+    fi
+}
+
+error() {
```
Comment (mxpv) [inline prepare-testdata.sh]: How this fn used?

Comment (kzys) [reply prepare-testdata.sh]: /dev/kvm check was there, and I forgot to remove the error function when I moved that from here to Makefile. Let me remove...

Comment (kzys) [reply prepare-testdata.sh]: https://github.com/firecracker-microvm/firecracker-go-sdk/pull/144/commits/3721c6919236d1c695c712e1bb082cf79bedb404 doesn't have the function.

File: Makefile [hunk 2]
```diff
@@ -13,18 +13,24 @@
...
-	go test -short ./... $(EXTRAGOARGS)
+unit-tests: test-setup
+	DISABLE_ROOT_TESTS=$(DISABLE_ROOT_TESTS) go test -short ./... $(EXTRAGOARGS)
 
-all-tests:
-	go test ./... $(EXTRAGOARGS)
+all-tests: test-setup
+	DISABLE_ROOT_TESTS=$(DISABLE_ROOT_TESTS) go test ./... $(EXTRAGOARGS)
+
+test-setup:
```
Comment (samuelkarp) [inline Makefile]: You can have make track the files so it knows whether or not to run this target each time.  Something like this:

```make
testdata_objects = testdata/vmlinux testdata/root-drive.img testdata/firecracker
test-setup: $(testdata_objects)
	@test -w /dev/kvm || \
		(echo "In order to run firecracker, $(shell whoami) must have write permission to /dev/kvm"; false)

$(testdata_objects):
	./prepare-testdata.sh
```

Comment (kzys) [reply Makefile]: If we use make to track the files, we need to pass the target to the shell script to only download the missing file.

```
$(testdata_objects):
	./prepare-testdata.sh $@
```

Another path we could take is having each target separately.

```
testdata/vmlinux:
        curl -fsSL -o $@ https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin

```

Comment (kzys) [reply Makefile]: Removed the shell script and used `make` for tracking existing files on https://github.com/firecracker-microvm/firecracker-go-sdk/pull/144/commits/c210e9bf1f8a2da501c21fe240463ab8b23bb3c8

File: Makefile [hunk 3]
```diff
@@ -13,18 +13,42 @@
...
+	$(curl) -o $@ https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin
+
+testdata/firecracker:
+	$(curl) -o $@ https://github.com/firecracker-microvm/firecracker/releases/download/v0.18.0/firecracker-v0.18.0
+	chmod +x $@
+
+testdata/root-drive.img:
+	$(curl) -o $@ https://s3.amazonaws.com/spec.ccfc.min/img/hello/fsfiles/hello-rootfs.ext4
+
+.PHONY: all generate clean build test unit-tests all-tests check-kvm
```
Comment (kzys) [inline Makefile]: distclean should be there...

**Pull request merged**